modules/gdscript/language_server/gdscript_extend_parser.cpp

```diff
@@ -0,0 +1,759 @@
+		}
+		symbol.documentation = parse_documentation_as_markdown(line);
+		r_symbol.children.push_back(symbol);
+	}
+}
+
+String ExtendGDScriptParser::marked_documentation(const String &p_bbcode) {
+
+	String markdown = p_bbcode.strip_edges();
+
+	Vector<String> lines = markdown.split("\n");
+	bool in_code_block = false;
+	int code_block_indent = -1;
+
+	markdown = "";
+	for (int i = 0; i < lines.size(); i++) {
+		String line = lines[i];
+		int block_start = line.find("[codeblock]");
+		if (block_start != -1) {
+			code_block_indent = block_start;
+			in_code_block = true;
+			line = "'''gdscript";
+			line = "\n";
+		} else if (in_code_block) {
+			line = "\t" + line.substr(code_block_indent, line.length());
+		}
+
+		if (in_code_block && line.find("[/codeblock]") != -1) {
+			line = "'''\n";
+			line = "\n";
+			in_code_block = false;
+		}
+
+		if (!in_code_block) {
+			line = line.strip_edges();
+			line = line.replace("[code]", "`");
+			line = line.replace("[/code]", "`");
+			line = line.replace("[i]", "*");
+			line = line.replace("[/i]", "*");
+			line = line.replace("[b]", "**");
+			line = line.replace("[/b]", "**");
+			line = line.replace("[u]", "__");
+			line = line.replace("[/u]", "__");
+			line = line.replace("[method ", "`");
+			line = line.replace("[member ", "`");
+			line = line.replace("[signal ", "`");
+			line = line.replace("[enum ", "`");
+			line = line.replace("[constant ", "`");
+			line = line.replace("[", "`");
+			line = line.replace("]", "`");
+		}
+
+		if (!in_code_block && i < lines.size() - 1) {
+			line += "\n\n";
+		} else if (i < lines.size() - 1) {
+			line += "\n";
+		}
+		markdown += line;
+	}
+	return markdown;
+}
+
+String ExtendGDScriptParser::parse_documentation(int p_line, bool p_docs_down) {
+	ERR_FAIL_INDEX_V(p_line, lines.size(), String());
+
+	List<String> doc_lines;
+
+	if (!p_docs_down) { // inline comment
+		String inline_comment = lines[p_line];
+		int comment_start = inline_comment.find("#");
+		if (comment_start != -1) {
+			inline_comment = inline_comment.substr(comment_start, inline_comment.length()).strip_edges();
+			if (inline_comment.length() > 1) {
+				doc_lines.push_back(inline_comment.substr(1, inline_comment.length()));
+			}
+		}
+	}
+
+	int step = p_docs_down ? 1 : -1;
+	int start_line = p_docs_down ? p_line : p_line - 1;
+	for (int i = start_line; true; i += step) {
+
+		if (i < 0 || i >= lines.size()) break;
+
+		String line_comment = lines[i].strip_edges(true, false);
+		if (line_comment.begins_with("#")) {
+			line_comment = line_comment.substr(1, line_comment.length());
+			if (p_docs_down) {
+				doc_lines.push_back(line_comment);
+			} else {
+				doc_lines.push_front(line_comment);
+			}
+		} else {
+			break;
+		}
+	}
+
+	String doc;
+	for (List<String>::Element *E = doc_lines.front(); E; E = E->next()) {
+		doc += E->get() + "\n";
+	}
+	return doc;
+}
+
+String ExtendGDScriptParser::get_text_for_completion(const lsp::Position &p_cursor) const {
+
+	String longthing;
```

Comment:
The `StringBuilder` class might be more suitable for this kind of string concatenation.